python/ql/src/Expressions/CallToSuperWrongClass.ql

```diff
@@ -13,17 +13,14 @@
  */
 
 import python
+import semmle.python.dataflow.new.DataFlow
 
-from CallNode call_to_super, string name
+from DataFlow::CallCfgNode call_to_super, string name
 where
-  exists(GlobalVariable gv, ControlFlowNode cn |
-    call_to_super = ClassValue::super_().getACall() and
-    gv.getId() = "super" and
-    cn = call_to_super.getArg(0) and
-    name = call_to_super.getScope().getScope().(Class).getName() and
-    exists(ClassValue other |
-      cn.pointsTo(other) and
-      not other.getScope().getName() = name
-    )
+  call_to_super.getFunction().getALocalSource().asExpr().(Name).getId() = "super" and
+  name = call_to_super.getScope().getScope().(Class).getName() and
+  exists(DataFlow::Node arg |
+    arg = call_to_super.getArg(0) and
+    not arg.getALocalSource().asExpr().(Name).getId() = name
```

Comment:
```suggestion
    arg.getALocalSource().asExpr().(Name).getId() != name
```
What you wrote is very broad, since it yields a result if, say, the local source for `arg` is not an instance of `Name`. I think it makes more sense to limit ourselves to the case where it _is_ a `Name`.

Reply:
Thanks! That may account for the many new results 😅

Reply:
That made quite a difference. The evaluation is now reasonable...